### PR TITLE
Setting User-Agent for Reverb API cURL requests

### LINKED ISSUE #39: Custom user-agent
Please set the user-agent on all curl requests to "Reverb-Magento". If possible append the current magento version as well, so it would look like "Reverb-Magento MagentoVersion=community-1.9"

If this code is duplicated, please extract to one place

### PR DESCRIPTION
**Pull request by dunagan5887**
Fix #39

### CODE REVIEW

File: app/code/community/Reverb/ReverbSync/Helper/Data.php [hunk 1]
```diff
@@ -281,6 +281,7 @@ protected function _getCurlResource($url, $options_array = array())
         $options_array[CURLOPT_SSL_VERIFYPEER] = 0;
         $options_array[CURLOPT_HEADER] = 0;
         $options_array[CURLOPT_RETURNTRANSFER] = 1;
+        $options_array[CURLOPT_USERAGENT] = 'Reverb-Magento MagentoVersion=community-1.9';
```
Comment (skwp) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: this is not supposed to be a hardcoded magento version. are we able to read the magento version off the instance?

Comment (dunagan5887) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: Ya, that would make more sense (facepalm) was kind of tired last night so wasn't thinking dynamically enough. I'll have this shortly

Comment (skwp) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: also whatever else we can glean from the  current instance would be very useful. for example its hosted domain (foo.com) so we can tell where the website is. if that's available, stick it in the user agen tplease

Comment (dunagan5887) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: What would be the format for this? 

'Reverb-Magento MagentoVersion=community-1.9 domain=foo.com';

As above?

Comment (skwp) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: MagentoVersion=whatever MagentoDomain=whatever

let's keep that consistent

Comment (dunagan5887) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: Sounds good. As a note, if there are instances where customers are on one domain, but the admin panel is on another domain, then requests coming as a result of order placements will show a different domain than ones caused by updates done via the admin panel, or by calls made via cron execution

Comment (skwp) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: actually can we get both of those values into custom headers as well
X-Magento-Version and X-Magento-Domain

Comment (skwp) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: Is there a good way to also include the version of the REverb plugin as a header? Where can we put the version so we can get something like "X-Magento-Reverb-Version: 0.3.3"

Comment (dunagan5887) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: For the Reverb Module package number I'll need to see the var/package/*.xml file associated with the extension. Could you grant me access to your Magento developer account? The email on my account is dunagan5887@gmail.com

Comment (dunagan5887) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: Currently the request is now looking as such

GET /api/my/listings?state=all&sku=new_curl HTTP/1.1
User-Agent: Reverb-Magento MagentoVersion=Community-1.9.2.0 MagentoDomain=reverb_magento.com
Host: sandbox.reverb.com
Accept: */*
X-Auth-Token: 1ea238014d948d056f189978d7ab837d1c8152112dfcf0d2eca2b2cfd53f7d3c
Content-type: application/hal+json
X-Magento-Version: Community-1.9.2.0
X-Magento-Domain: reverb_magento.com

Does this look correct?

**Comment by dunagan5887** [comment]
Fix #42 Fix #39 

I guess this hasn't been merged in yet, this branch now has the PUT issue fix as well as the dynamic User-Agent headers (does not include the Reverb-Magento module package version)

**Pull request merged**